uaclient/entitlements/repo.py

```diff
@@ -173,10 +174,14 @@ def setup_apt_config(self) -> None:
                 'No specific resourceToken present. Using machine token'
                 ' as %s credentials', self.title)
             token = self.cfg.machine_token['machineToken']
-        if directives.get('aptKey'):
-            logging.debug(
-                "Ignoring aptKey directive '%s'", directives.get('aptKey'))
-        keyring_file = os.path.join(apt.KEYRINGS_DIR, self.repo_key_file)
+        aptKey = directives.get('aptKey')
+        if not aptKey:
+            logging.warning(
+                "No aptKey directive provided for %s", self.name)
+        else:
+            keyring_file = os.path.join(apt.APT_KEYS_DIR, self.repo_key_file)
+            gpg.export_gpg_key_from_keyring(
```

Comment:
Key installation was previously performed by `add_auth_apt_repo`; what's the thinking behind moving it up a layer here?

Reply:
I moved it up for two reasons:
 - it needs self.repo_key_file from the Entitlement to generate the target keyring_file path and aptKey from self.cfg directives. Instead of passing two params into add_auth_apt_repo to do one thing, I thought we could make the call up here and simplify add_auth_apt_repo call signature.

Reply:
IMO, it would be preferable to have the apt function be "atomic" even if it requires another parameter.

(That only looks like one reason though, so perhaps there's more to it?)

Reply:
When it was a simple file copy, it made sense to do that inside add_auth_apt_repo. Now that we have to do a gpg key extraction and then a copy to a named keyring_file destination that makes ad_auth_apt_repo grow another parameter in the function signature, one for the key_id and one for the destination keyring file to write. I thought it'd be simpler to reduce the number of params and do the simple work here, but if you prefer the other way, I can extend add_auth_apt_repo.

Reply:
Will fix that to accept the additional param.